neqo-bin/src/client/mod.rs

```diff
@@ -351,6 +351,7 @@ trait Handler {
 
     fn handle(&mut self, client: &mut Self::Client) -> Res<bool>;
     fn take_token(&mut self) -> Option<ResumptionToken>;
+    fn reinit(&mut self);
```

Comment:
`reinit` is always called on the concrete h09 or h3 Handler, never on the `Handler` trait, right? If so, I suggest removing the trait extension here and instead implement `reinit` on the concrete Handler types (e.g. `impl <'a> Handler<'a> { fn reinit(&mut self) {`) instead.

Not a strong opinion.